torch/_functorch/aot_autograd.py

```diff
@@ -1019,7 +1019,8 @@ def fn_to_trace(*args):
     # Next, the input args
     full_args.extend(args)
 
-    with ctx():
+    from torch.export.safeguard import GradModeUnsupportedSafeguard
+    with ctx(), GradModeUnsupportedSafeguard():
```

Comment:
The reason that I don't put the safeguard ctx at the early place (such as https://github.com/pytorch/pytorch/blob/65c5eed01d3bf1b971f97c143beaa9dd8b0be0df/torch/export/_trace.py#L370), because `torch.no_grad` could be possibly invoked here and the safeguard will capture and error, which is not expected.